app/src/main/java/com/udacity/nanodegree/blooddonation/ui/login/view/UserLoginActivity.java

```diff
@@ -52,6 +63,14 @@
     mPresenter.onCreate();
   }
 
+  public void checkForValidPhoneNumber(){
+
+    etPhoneNumber = ((ActivityUserLoginBinding) mBinding).etPhoneNumber;
+    iAmInButton = ((ActivityUserLoginBinding) mBinding).bvIn;
+
+    AppUtil.validatePhoneNumber(etPhoneNumber , iAmInButton);
```

Comment:
Why pass view in validatePhoneNumber? Why not just pass the count and text and have listener here itself.

validatePhoneNumber(int count,String text) -> let this function return true or false based on your validation logic.